### ISSUE TITLE
Control Tag Filtering UI: Accomodate for Key Pair Values without values defined

### ISSUE THREAD

**Issue by jonong1972** [chef/automate#1713]
## User Story

As a Automate User, I will use this work when I want to filter a control tag when it has no values associated to it.

## Overview
We need to accommodate for Control Tags (keys) that do not have any values within the filter component. 

There may be some logic to be added to the filter component to accommodate for this. To detect if a Control Tag has any values. If it does not, it essentially transforms the filter into a Boolean search to see which items have that tag.

## Demo Script / Repro Steps
1. Goto Compliance Reports page
2. in the filter box select `Control Tags`
3. System will provide suggestions for Control Tags
4. Choose a control tag (for this test, a control tag that does not have any values)
5. System will determine if the control tag has a value or not associated with it.
6.a If the control tag does not: System will then initiate filter and add the tag
6.b if the control tag does: System will then suggest the values.

### Badge for Key Pair Value when no Value is defined
![Screen Shot 2019-09-25 at 11 17 40 AM](https://user-images.githubusercontent.com/1316587/65628391-39369180-df86-11e9-8314-19b5dcfbfa0e.png)

### Badge for Key Pair Value when a Value is defined
![Screen Shot 2019-09-25 at 11 17 46 AM](https://user-images.githubusercontent.com/1316587/65628404-3cca1880-df86-11e9-90c0-4c0c823b8b62.png)

**Comment by alexpop** [comment]
@scottopherson we looked into this today in standup and we have two main ways to accomplish this:

1. Update the suggestions API to provide a special value when a control tag key has no value. Based on this the UI can display the right text to the user. I prefer not to do much magic in the backend because it can get confusing and inconsistent over time.

2. Provide some default suggestions options that are not related to the content. These can be instant and not have to wait for the backend to provide the possible values. For example, one option to filter for a key with no value and second to filter for a key with any value. Or the user selects one of the suggested values the backend returns. This is what the AWS EC2 dashboard is doing when filtering by tags:


<img width="432" alt="Screen Shot 2019-10-24 at 8 19 26 PM" src="https://user-images.githubusercontent.com/107378/67518518-d592a780-f69c-11e9-9f6c-1e1729f76960.png">

<img width="553" alt="Screen Shot 2019-10-24 at 8 20 09 PM" src="https://user-images.githubusercontent.com/107378/67518517-d592a780-f69c-11e9-98c8-793b6d41e8a6.png">

**Comment by scottopherson** [comment]
@alexpop option 2 sounds like a great idea. Having an empty option the user can explicitly select instead of waiting on API response and automagically 'selecting' an empty option for them based on that response would solve the original issue nicely I think.

**Comment by alexpop** [comment]
Great thanks for confirming Scott. I'll work on the API to ensure we have a filter for control tag with empty/no value.

**Comment by vjeffrey** [comment]
ok, alex added changes to allow filtering by "no values" for a control tag value (https://github.com/chef/automate/pull/1989)

i pulled the change down to play around with it.

so here's what we need to do to complete this work:

- for control tag value suggestions only, hardcode a "no value" field that will always be at the top of the suggestions list 
- when a user selects the no value field, the api call should look like this:
```
curl --silent --insecure -H "api-token: value" https://a2-dev.test/api/v0/compliance/reporting/nodes/search -d '{"filters":[{"type": "control_tag:ssh", "values": [""]}] }'
```
- since we or between values for a control tag key, a user could ask to see all nodes that ran a control tag that has no values + all nodes that ran a control tag that has x value. for example:
```
curl --silent --insecure -H "api-token: value" https://a2-dev.test/api/v0/compliance/reporting/nodes/search -d '{"filters":[{"type": "control_tag:ident", "values": ["", "CCA*"]}] }'
```

**Comment by SEAjamieD** [comment]
@vjeffrey - While working on this, I noticed that the `Clear All` button does not work when filtering by control tags.  I've confirmed that this is the case on Master as well.  I'm about to open a PR for filter by no values right now and think that fixing the clear all button should be a new ticket.

**Comment by SEAjamieD** [comment]
> @vjeffrey - While working on this, I noticed that the `Clear All` button does not work when filtering by control tags. I've confirmed that this is the case on Master as well. I'm about to open a PR for filter by no values right now and think that fixing the clear all button should be a new ticket.

Ticket opened: https://github.com/chef/automate/issues/2167
